ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -605,12 +604,37 @@ const DataSetPage = fp.flow(withUserProfile(), withCurrentWorkspace(), withUrlPa
       this.setState({selectedDomainValuePairs: valuesSelected, dataSetTouched: true});
     }
 
-    get selectAll() {
-      return fp.isEmpty(this.state.selectedDomainValuePairs);
+    // Append newValueSets (values from API) to state's valueSets and to selectedDomainValuePairs.
+    // Set valuesLoading to false once the state is updated
+    updateValueSets(newValueSets) {
```

Comment:
(Non-opt) When possible, prefer to annotate the type on any method parameters. Even if TypeScript can infer this, it's helpful for the reader to know which type to keep in mind when reading through this function.